src/test/java/com/restapi/carMarket/service/CarServiceUnitTest.java

```diff
@@ -11,168 +8,199 @@
     }
 
     @Test(expected = CarNotValidException.class)
-    public void insertionOfCarWithPriceToHighShouldThrowCarNotValidException() throws Exception{
+    public void insertionOfCarWithPriceToHighShouldThrowCarNotValidException() {
         Car car = new Car("BMW","5 Series", 2010, 2100000000);
-        carService.insert(car);
+        carService.addCarToMarket(car);
+        Mockito.verify(carDAO,Mockito.times(0)).save(car);
+    }
+
+    @Test(expected = CarNotValidException.class)
+    public void insertionOfCarWithBlankBrandShouldThrowCarNotValidException() {
+        Car car = new Car("","5 Series", 2010, 2100000);
+        carService.addCarToMarket(car);
+        Mockito.verify(carDAO,Mockito.times(0)).save(car);
+    }
+
+    @Test(expected = CarNotValidException.class)
+    public void insertionOfCarWithNullBrandShouldThrowCarNotValidException() {
+        Car car = new Car(null,"5 Series", 2010, 2100000);
+        carService.addCarToMarket(car);
+        Mockito.verify(carDAO,Mockito.times(0)).save(car);
+    }
+
+    @Test(expected = CarNotValidException.class)
+    public void insertionOfCarWithBlankModelShouldThrowCarNotValidException() {
+        Car car = new Car("BMW","", 2010, 2100000);
+        carService.addCarToMarket(car);
+        Mockito.verify(carDAO,Mockito.times(0)).save(car);
+    }
+
+    @Test(expected = CarNotValidException.class)
+    public void insertionOfCarWithNullModelShouldReturnCarNotValidException() {
+        Car car = new Car("BMW",null, 2010, 2100000);
+        carService.addCarToMarket(car);
         Mockito.verify(carDAO,Mockito.times(0)).save(car);
     }
 
     @Test
-    public void findAllCarsShouldCallFindAllMethodOfDAO() throws Exception{
-        carService.findAll();
+    public void findAllCarsShouldCallFindAllMethodOfDAO() {
+        carService.getAllCarsOnMarket();
         Mockito.verify(carDAO,Mockito.times(1)).findAll();
     }
 
     @Test
-    public void findExistentCarByIdShouldReturnThisCar() throws Exception{
+    public void findExistentCarByIdShouldReturnThisCar() {
         Car car = new Car("BMW","5 Series", 2200, 40000);
         Optional<Car> returnedCar = Optional.of(car);
         Long id = 10L;
 
         given(carDAO.findById(id)).willReturn(returnedCar);
-        Car newCar = carService.findById(id);
+        Car newCar = carService.getCarById(id);
 
         Assert.assertTrue(newCar.equals(car));
         Mockito.verify(carDAO,Mockito.times(1)).findById(id);
     }
 
     @Test(expected = CarNotFoundException.class)
-    public void findNonExistentCarByIdShouldThrowCarNotFoundException() throws Exception{
+    public void findNonExistentCarByIdShouldThrowCarNotFoundException() {
         Optional<Car> returnedCar = Optional.empty();
         Long id = 10L;
 
         given(carDAO.findById(id)).willReturn(returnedCar);
-        carService.findById(id);
+        carService.getCarById(id);
 
         Mockito.verify(carDAO,Mockito.times(1)).findById(id);
```

Comment:
No need to verify here. We already verified that the dao is called by the previous test case (line 112). Between these two test cases it's very clear that method call outcome is different based on the return value of the call to the dao, therefore we know it's called.

Reply:
Done